### PR TITLE
Add method to get current fetch amount to KafkaConsumer

### PR DESCRIPTION
**Pull request by pcasaes**
Signed-off-by: Paulo Casaes <pcasaes@gmail.com>

Motivation:

On the smallrye-reactive-messaging project we are using the pause/resume (no parameters) methods when configure a consumer re-balance listener. The basic design is: pause when there's a re-balance and only resume after the listener has completed its work. We do this so as not to receive messages while waiting on an eventual asynchronous seek the listener may perform - a very common use case for the kafka consumer re-balance listener is to seek to specific offsets (see Kafka: The Definitive Guide page 85).

The particular problem we are having is described in this PR
https://github.com/smallrye/smallrye-reactive-messaging/pull/672

Basically it's not suppose to go to flowing mode which is what **resume** does, it's suppose to remain in **fetch** mode. To facilitate solving this particular problem it would be nice to have a way to know the current fetch amount.

### CODE REVIEW

File: src/main/java/io/vertx/kafka/client/consumer/KafkaReadStream.java [hunk 1]
```diff
@@ -69,6 +69,18 @@
...
+   * Returns the current fetch amount.
+   *
+   * <ul>
+   *   <i>If the stream is in <i>flowing</i> mode will return {@link Long#MAX_VALUE}.</i>
+   *   <li>If the stream is in <i>fetch</i> mode, will return the current number of elements still to be delivered or 0 if paused.</li>
+   * </ul>
+   *
+   * @return current fetch amount
+   */
+  long currentFetchAmount();
```
Comment (pcasaes) [inline src/main/java/io/vertx/kafka/client/consumer/KafkaReadStream.java:82]: Another approach would be to return `OptionalLong` since `Long.MAX_VALUE` for *flowing* mode is an implementation detail. In this case we can return `empty` if it's in *flowing* mode.

Comment (vietj) [reply src/main/java/io/vertx/kafka/client/consumer/KafkaReadStream.java:82]: rather return rather Long , optional will not work well with Vert.x code generator

Comment (vietj) [reply src/main/java/io/vertx/kafka/client/consumer/KafkaReadStream.java:82]: that being said doing fetch(Long.MAX_VALUE) and resume() should work the same, so returning Long.MAX_VALUE makes sense as your code does not need if/else statements

File: src/main/java/io/vertx/kafka/client/consumer/KafkaConsumer.java [hunk 2]
```diff
@@ -176,6 +176,18 @@
...
+   * Returns the current fetch amount.
+   *
+   * <ul>
+   *   <i>If the stream is in <i>flowing</i> mode will return {@link Long#MAX_VALUE}.</i>
+   *   <li>If the stream is in <i>fetch</i> mode, will return the current number of elements still to be delivered or 0 if paused.</li>
+   * </ul>
+   *
+   * @return current fetch amount
+   */
+  long currentFetchAmount();
```
Comment (vietj) [inline src/main/java/io/vertx/kafka/client/consumer/KafkaConsumer.java:189]: This should be named `demand()`, the fetch is the value added to the current demand to produce a new demand value

File: src/main/java/io/vertx/kafka/client/consumer/KafkaReadStream.java (hunk 1 above)
Comment (pcasaes) [reply src/main/java/io/vertx/kafka/client/consumer/KafkaReadStream.java:82]: Ok

File: src/main/java/io/vertx/kafka/client/consumer/KafkaConsumer.java (hunk 2 above)
Comment (pcasaes) [reply src/main/java/io/vertx/kafka/client/consumer/KafkaConsumer.java]: done

**Comment by pcasaes** [comment]
Is it possible to get this into 3.9?

**Comment by vietj** [comment]
that will require a backport PR

**Comment by pcasaes** [comment]
Cool. Do I just rebase this PR to the 3.9 branch or do we open a new PR cherry picking against the 3.9 branch?

**Comment by ppatierno** [comment]
Thanks for the PR, if @vietj doesn't see any problem with just doing a cherry-pick on 3.9 I can do that.
I don't see this PR using stuff only in Vert.x 4.x that cannot be just cherry picked to 3.9. Wdyt @vietj ?

**Comment by pcasaes** [comment]
Thanks

**Comment by pcasaes** [comment]
Also. I originally tested it against 3.9.2

**Comment by vietj** [comment]
Go ahead with merge and back port @ppatierno

**Comment by ppatierno** [comment]
Cherry-picked to 3.9 branch.
